src/index.ts

```diff
@@ -50,7 +51,20 @@ async function doctor(projectPathArg: string, options: DoctorCommandOptions) {
   Context.create(project);
 
   await project.loadFilesContent();
-  await project.runAllChecks();
+
+  // Run diagnostics first
+  await project.runChecks(CheckGroup.Diagnostics);
+  // Run other checks in order
+  const checkGroups: CheckGroup[] = [
+    CheckGroup.Initialization,
+    CheckGroup.PushSetup,
+    CheckGroup.Dependencies,
+  ];
```

Comment:
```suggestion
  const checkGroups: CheckGroup[] = [
    CheckGroup.Diagnostics
    CheckGroup.Initialization,
    CheckGroup.PushSetup,
    CheckGroup.Dependencies,
  ];
```

Is this code above doing the same thing? I don't understand why CheckGroup.Diagnostics is separate?

Reply:
To match the exact output that we are expecting.

Output with current implementation (CheckGroup.Diagnostics outside the array)
```
Starting diagnostic for React Native project...
[✓] Project Path: [path]
[✓] Customer.io React Native SDK version: [SDK Version]

Initialization
[✓] React Native SDK Initialization found
```

Output with suggested implementation (CheckGroup.Diagnostics inside the array)

```
Starting diagnostic for React Native project...
[✓] Project Path: [path]

Diagnostics
[✓] Customer.io React Native SDK version: [SDK Version]

Initialization
[✓] React Native SDK Initialization found
```

Reply:
Hmmm. By reading the code, that is not clear to me. 

I am OK with the code as-is because there are comments suggesting not to include Diagnostics in the array. I also don't currently have ideas to suggest to improve it at the moment. 

👍